### ISSUE TITLE
ArgumentError in `Sibyl.Handlers.attach_all_events/2` in docker

### ISSUE THREAD

**Issue by florius0** [Vetspire/sibyl#30]
I was evaluating Sybil and came across this error:

```elixir
** (Mix) Could not start application fish_finder: exited in: FF.Application.start(:normal, [])
     ** (EXIT) an exception was raised:
         ** (ArgumentError) errors were found at the given arguments:

   * 1st argument: invalid destination

             :erlang.send(:undefined, {:shell_state, #PID<0.3858.0>})
             (stdlib 5.0.1) shell.erl:590: :shell.get_state/0
             (stdlib 5.0.1) shell.erl:597: :shell.get_function/2
             (stdlib 5.0.1) shell_default.erl:159: :shell_default."$handle_undefined_function"/2
             (sibyl 0.1.2) lib/sibyl/events.ex:105: Sibyl.Events.reflect/1
             (elixir 1.15.0) lib/enum.ex:4317: Enum.flat_map_list/2
             (elixir 1.15.0) lib/enum.ex:4318: Enum.flat_map_list/2
             (sibyl 0.1.2) lib/sibyl/handlers.ex:20: Sibyl.Handlers.attach_all_events/2
             (fish_finder 0.1.0) lib/fish_finder/application.ex:29: FF.Application.start/2
             (kernel 9.0.1) application_master.erl:293: :application_master.start_it_old/4
```

It was not reproduceable on my local elixir, but only in docker.

[My repo using which the bug may be reproduced.](https://github.com/florius0/elixir_grafana_loki_tempo/tree/telemetry)

From what I'm gathering, `:shell.__info__` results not in exception, but in message sending to `:undefied` pid.

## Versions

Local: 

 - OS: macOS 13.4
 - Erlang/Elixir:
  ```
  Erlang/OTP 25 [erts-13.2.2.1] [source] [64-bit] [smp:12:12] [ds:12:12:10] [async-threads:1] [jit:ns] [dtrace]
  
  Elixir 1.15.0 (compiled with Erlang/OTP 25)
  ```

Docker:

  - Runtime: Orbstack 0.16.1
  - `bitwalker/alpine-elixir-phoenix:1.15.0`

**Comment by vereis** [comment]
Beautiful, thanks for the reproduction steps.

Will take a look and get back to you soon :crossed_fingers:

**Comment by florius0** [comment]
I might be able to fix it today or narrow the problem at least

**Comment by florius0** [comment]
Using [`module_info/2`](https://github.com/florius0/sibyl/blob/8e4b2e491cf01829ddce58c16c1d1b8c7d4d1c34/lib/sibyl/events.ex#L104) instead of `__info__/2` in `Sibyl.Events.reflect/1` seems have fixed the issue.

In my fork I've updated to Elixir 1.15.0 (tho I'm not sure yet that it didn't caused any problems), and renamed guards and predicates to match elixir conventions (`is_...` for guards, `...?` for predicates).

If you would like I can include those changes in PR.

PS
I would like to suggest to run your CI on all minor versions from lowest supported up to latest

**Comment by vereis** [comment]
@florius0 that sounds awesome! Looking forward to seeing a PR :raised_hands: Thank you!

And yes, running CI against a matrix of supported versions (defining what those are explicitly) will be great too, thank you for the suggestion!

**Comment by florius0** [comment]
Tried it on `elixir:1.15.4-otp-25-alpine`, same issue. Perhaps it has to do smth with alpine

**Comment by rob-vetspire** [comment]
I can confirm issue https://github.com/Vetspire/sibyl/issues/30 also occurs when running an application with `elixir 1.15.7-otp-26` and `erlang 26.1.2` directly on `macos`.

**Comment by florius0** [comment]
I guess it is related to new stripping of unused functions by elixir compiler maybe...

**Comment by vereis** [comment]
@florius0 after investigating, I think we should just skip all Erlang modules when doing reflection, as reflection will only discover events defined by `Sibyl`, and that can only happen in Elixir applications.

I think this is probably a better fix than relying on `module.module_info`, as at the very least, the `__info__/1` callback is documented: https://hexdocs.pm/elixir/1.7.3/Module.html

Have the tentative fix in https://github.com/Vetspire/sibyl/pull/34, will just need to do some regression testing but I believe this fixes the problem!

**Comment by florius0** [comment]
I don't think that restricting Erlang compatibility is a good option especially when there is no cost of keeping it.

If my memory serves me, `Sybil` defines events as module attributes (with `persist: true`), and getting such attribute with `module_info/1` is ok.
It is not documented in Elixir docs since it is Erlang callback: https://www.erlang.org/doc/reference_manual/modules#module_info-0-and-module_info-1-functions.

As for delay, do not worry, we use my fork for the time being.

**Comment by vereis** [comment]
Ah I wasn't aware of that. Will revisit my change, thanks ❤️

**Comment by vereis** [comment]
Resolved in #34
